did/peer.go

```diff
@@ -0,0 +1,539 @@
+	return PeerDelta{}, errors.Wrap(util.NotImplementedError, "peer:did delta")
+}
+
+// TODO: CRDT
+// https://identity.foundation/peer-did-method-spec/#crdts
+
+// Generates the key by types
+func (d DIDPeer) generateKeyByType(kt crypto.KeyType) (gocrypto.PublicKey, gocrypto.PrivateKey, error) {
+	if !isSupportedKeyType(kt) {
+		err := fmt.Errorf("%s : %s for did:peer", util.UnsupportedError, kt)
+		return nil, nil, err
+	}
+	return crypto.GenerateKeyByKeyType(kt)
+}
+
+func (m PeerMethod0) Generate(kt crypto.KeyType, publicKey gocrypto.PublicKey) (*DIDPeer, error) {
+	var did DIDPeer
+	encoded, err := encodePublicKeyWithKeyMultiCodecType(kt, publicKey)
+	if err != nil {
+		return nil, errors.Wrap(err, "could not encode public key for did:peer")
+	}
+	did = buildDIDPeerFromEncoded(0, encoded)
+	return &did, err
+}
+
+// Method 1: genesis doc
+type PeerMethod1 struct{}
+
+// Method 2: multiple inception key without doc
+type PeerMethod2 struct {
+	KT     crypto.KeyType
+	Values []interface{}
+}
+
+// https://identity.foundation/peer-did-method-spec/index.html#generation-method
+const (
+	PeerPurposeEncryptionCode           PurposeType = "E"
+	PeerPurposeAssertionCode            PurposeType = "A"
+	PeerPurposeVerificationCode         PurposeType = "V"
+	PeerPurposeCapabilityInvocationCode PurposeType = "I"
+	PeerPurposeCapabilityDelegationCode PurposeType = "D"
+	PeerPurposeCapabilityServiceCode    PurposeType = "S"
+)
+
+// For a quick lookup of supported DID purposes
+var supportedDIDPeerPurposes = map[PurposeType]bool{
+	PeerPurposeEncryptionCode:           true,
+	PeerPurposeAssertionCode:            true,
+	PeerPurposeVerificationCode:         true,
+	PeerPurposeCapabilityDelegationCode: true,
+	PeerPurposeCapabilityInvocationCode: true,
+	PeerPurposeCapabilityServiceCode:    true,
+}
+
+func (d DIDPeer) IsValidPurpose(p PurposeType) bool {
+	if _, ok := supportedDIDPeerPurposes[p]; ok {
+		return true
+	}
+	return false
+}
+
+func (m PeerMethod0) Resolve(did DID) (*DIDDocument, error) {
+
+	d, ok := did.(DIDPeer)
+	if !ok {
+		return nil, errors.Wrap(util.CastingError, "did:peer")
+	}
+
+	v, err := d.Parse()
+	if err != nil {
+		return nil, err
+	}
+
+	pubKey, keyType, err := decodeEncodedKey(v)
+	if err != nil {
+		return nil, err
+	}
+
+	keyReference := Hash + v
+	id := string(d)
+
+	verificationMethod, err := constructVerificationMethod(id, keyReference, pubKey, keyType)
+	if err != nil {
+		return nil, err
+	}
+
+	verificationMethodSet := []VerificationMethodSet{
+		[]string{keyReference},
+	}
+
+	return &DIDDocument{
+		Context:              KnownDIDContext,
+		ID:                   id,
+		VerificationMethod:   []VerificationMethod{*verificationMethod},
+		Authentication:       verificationMethodSet,
+		AssertionMethod:      verificationMethodSet,
+		KeyAgreement:         verificationMethodSet,
+		CapabilityDelegation: verificationMethodSet,
+	}, nil
+}
+
+func (m PeerMethod1) Resolve(d DID) (*DIDDocument, error) {
```

Comment:
ah I see here it's marked as not implemented. seems useful for handling method 1 dids we may encounter